deltacat/compute/compactor_v2/compaction_session.py

```diff
@@ -98,14 +105,26 @@ def compact_partition(params: CompactPartitionParams, **kwargs) -> Optional[str]
             f"Partition-{params.source_partition_locator} -> "
             f"Compaction session data processing completed"
         )
-        round_completion_file_s3_url = None
+        round_completion_file_s3_url: Optional[str] = None
         if new_partition:
-            logger.info(f"Committing compacted partition to: {new_partition.locator}")
+            previous_partition: Optional[Partition] = None
+            if is_inplace_compacted:
+                previous_partition: Optional[
```

Comment:
Perfect! I was thinking of this case and now you have already resolved it. The partition ID we get may be stale by the time commit_partition is called. Can we add documentation on why we need to get again?